src/trajectory.jl

```diff
@@ -668,79 +594,63 @@ function generalised_uturn_criterion(rho, p_sharp_minus, p_sharp_plus)
     return (dot(rho, p_sharp_minus) <= 0) || (dot(rho, p_sharp_plus) <= 0)
 end
 
-function isterminated(h::Hamiltonian, t::BinaryTree{T}, args...) where {T<:Union{ClassicNoUTurn, GeneralisedNoUTurn}}
-    return isterminated(h, t)
-end
-
 """
 Recursivly build a tree for a given depth `j`.
 """
-function build_tree(
-    rng::AbstractRNG,
-    nt::NUTS{S,C,I,F},
-    h::Hamiltonian,
-    z::PhasePoint,
-    sampler::AbstractTrajectorySampler,
-    v::Int,
-    j::Int,
-    H0::AbstractFloat,
-) where {I<:AbstractIntegrator,F<:AbstractFloat,S<:AbstractTrajectorySampler,C<:AbstractTerminationCriterion}
+function build_tree(rng, int, tc, h, z, sampler, v, j, H0)
     if j == 0
         # Base case - take one leapfrog step in the direction v.
-        z′ = step(nt.integrator, h, z, v)
+        z′ = step(int, h, z, v)
```

Comment:
Consider keeping previous variable name: `integrator` is a lot more readable than `int`.

Reply:
Sure.